File: backend-old/app/models/transaction.py
from pydantic import BaseModel, validator
from typing import Optional

class TransactionRequest(BaseModel):
    card_number: str
    amount: float
    merchant_id: str

    @validator("card_number")
    def validate_card_number(cls, value):
        if not value.isdigit() or len(value) != 16:
            raise ValueError("Invalid card number")
        return value

    @validator("amount")
    def validate_amount(cls, value):
        if value <= 0:
            raise ValueError("Amount must be greater than 0")
        return value

class TransactionResponse(BaseModel):
    success: bool
    message: str
    data: dict
    response_code: Optional[str] = None
    authorization_code: Optional[str] = None

File: backend/src/models/Transaction.js
const Joi = require('joi');

// Transaction request validation schema
const transactionRequestSchema = Joi.object({
  card_number: Joi.string()
    .pattern(/^\d{16}$/)
    .required()
    .messages({
      'string.pattern.base': 'Invalid card number'
    }),
  
  amount: Joi.number()
    .positive()
    .required()
    .messages({
      'number.positive': 'Amount must be greater than 0'
    }),
  
  merchant_id: Joi.string()
    .required()
});

// Transaction response structure
class TransactionResponse {
  constructor({ success, message, data, response_code = null, authorization_code = null }) {
    this.success = success;
    this.message = message;
    this.data = data;
    this.response_code = response_code;
    this.authorization_code = authorization_code;
  }
}

module.exports = {
  transactionRequestSchema,
  TransactionResponse,
  validateTransactionRequest: (data) => transactionRequestSchema.validate(data)
};